### PR TITLE
worker: add metrics

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
part of #543 

### What is changed and how it works?
add/modify two types of metrics:
- add label `source_id` for state of subtasks
- add counter of number of operate errors by (worker, error type), remove all `TODO: add better metrics`

### Check List 

Tests 

 - pass original tests

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Side effects

 - Possible performance regression

Related changes

### CODE REVIEW

File: dm/worker/metrics.go [hunk 1]
```diff
@@ -123,6 +137,8 @@ func InitStatus(lis net.Listener) {
 	}
 }
 
-func (st *SubTask) removeLabelValuesWithTaskInMetrics(task string) {
-	taskState.DeleteAllAboutLabels(prometheus.Labels{"task": task})
+func (st *SubTask) removeLabelValuesWithTaskInMetrics(task string, source string) {
+	taskState.DeleteAllAboutLabels(prometheus.Labels{"task": task, "source_id": source})
 }
+
+// opErrCounter cleans on worker close, which is the same time dm-worker exits, so no explicit clean
```
Comment (WangXiangUSTC) [inline dm/worker/metrics.go:144]: 😕

File: dm/worker/worker.go [hunk 2]
```diff
@@ -330,11 +333,16 @@ func (w *Worker) resetSubtaskStage(etcdCli *clientv3.Client) (int64, error) {
 			// TODO: right operation sequences may get error when we get etcdErrCompact, need to handle it later
 			// For example, Expect: Running -(pause)-> Paused -(resume)-> Running
 			// we get an etcd compact error at the first running. If we try to "resume" it now, we will get an error
-			err = w.operateSubTaskStage(stage, subtaskCfg)
+			op, err := w.operateSubTaskStage(stage, subtaskCfg)
```
Comment (WangXiangUSTC) [inline dm/worker/worker.go:336]: how about return the operate type string for these functions, so don't need to judge whether `op` is nil.

File: dm/worker/metrics.go [hunk 3]
```diff
@@ -34,14 +34,27 @@ import (
...
 	taskState = metricsproxy.NewGaugeVec(
 		prometheus.GaugeOpts{
 			Namespace: "dm",
 			Subsystem: "worker",
 			Name:      "task_state",
 			Help:      "state of task, 0 - invalidStage, 1 - New, 2 - Running, 3 - Paused, 4 - Stopped, 5 - Finished",
-		}, []string{"task"})
+		}, []string{"task", "source_id"})
+
+	opErrCounter = metricsproxy.NewCounterVec(
```
Comment (WangXiangUSTC) [inline dm/worker/metrics.go:51]: how to use this metric? send an alert message when the value is changed?

Comment (lance6716) [reply dm/worker/metrics.go:51]: show in grafana for example, user could thus know worker's state and decide to retire this worker or add more worker.

File: dm/worker/metrics.go (hunk 1 above)
Comment (lance6716) [reply dm/worker/metrics.go:144]: should I delete this comment? or move it to elsewhere

Comment (WangXiangUSTC) [reply dm/worker/metrics.go:144]: no code under this comment 😄

**Pull request merged**